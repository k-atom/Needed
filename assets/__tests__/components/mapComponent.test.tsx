import * as React from 'react';
import { mount } from 'enzyme';

import MapComponent from '../../ts/components/mapComponent';

let lat: number = 25.032697;
let lng: number = 121.563086;
let position:[number, number] = [lat, lng];
let zoom: number = 13;

const setLat = (num: number) => {
  lat = num;
  setPosition([lat, lng]);
}

const setLng = (num: number) => {
  lng = num;
  setPosition([lat, lng]);
}

const setPosition = ([lat, lng]: [number, number]) => {
  position = [lat, lng];
  previous.position = position;
}

const setZoom = (num: number) => {
  zoom = num;
}

let previous = {
  lat: lat,
  lng: lng,
  position: position,
  zoom: zoom,
  fn: {
    setLat: setLat,
    setLng: setLng,
    setPosition: setPosition,
    setZoom: setZoom
  }
}

describe('MapComponent', () => {
  const component = mount(<MapComponent {...previous}/>);

  it('Check node of MapComponent', () => {
    expect(component.find('div.mapComponent')).toHaveLength(1);
    expect(component.find('Button')).toHaveLength(1);
    expect(component.find('i.material-icons')).toHaveLength(1);
    expect(component.find('i.material-icons').text()).toEqual('my_location');
    expect(component.html()).toMatchSnapshot();
  });
  it('Should be get geoLocation', () => {
    component.find('Button').props().onClick({} as any);
  });
  it('Should not get geoLocation', () => {
    component.find('Button').props().onClick({} as any);
  });
  it('Should not create geoLocation', () => {
    (window.navigator.geolocation as any) = undefined;
    component.find('Button').props().onClick({} as any);
  });
});
