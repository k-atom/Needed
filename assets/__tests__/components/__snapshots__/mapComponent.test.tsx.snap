// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapComponent Check node of MapComponent 1`] = `"<div class=\\"mapComponent bottom right\\"><button title=\\"定位當前位置\\" type=\\"button\\" class=\\"ant-btn btn\\"><i class=\\"material-icons\\">my_location</i></button></div>"`;
